frontend/packages/client/src/components/WalletConnect.js

```diff
@@ -62,6 +67,11 @@ const SignInOutButton = ({
     return () => clearTimeout(timeout);
   }, [addressCopied]);
 
+  useEffect(() => {
+    setUserID(addr);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [addr]);
```

Comment:
this seems a little odd - why not just pass `addr` to `useNotificationService()`? I dont think we need to fetch arbitrary wallet addresses yet, just the current connected wallet.

Reply:
Good question, the reason is exactly that: we only care about current connected wallet, when instantiated, `useNotificationService` reads from the context, we don't need to pass the address as part of the configuration. If it helps, consider the following example:

- We want to build a hook to produce apple, 
```
const useProduceApple = () =>{
    const seed = () =>{
        console.log("seeding apple")
    }
    const grow = ()=>{
      console.log("growing apple")
    }
    const harvest = ()=>{
        console.log("harvesting apple")
    }
    return {
        seed,
        grow,
        harvest
    }
}
```
- Now we want to grow banana too (the structure you are referring to)
```
const useProduceFruit = (fruitName) =>{
    const seed = () =>{
        console.log("seeding" + fruitName)
    }
    const grow = ()=>{
      console.log("growing" + fruitName)
    }
    const harvest = ()=>{
        console.log("harvesting" + fruitName)
    }
    return {
        seed,
        grow,
        harvest
    }
}
```

Reply:
I think we don't need to set the `addr` this way, the SignInOutButton component is using a HOC, at the end of the file is wrapped by Web3Consumer. 

`export default Web3Consumer(CurrentUser);`
so addr is passed from the Web3Consumer to the SignInOutButton and inside we are calling the useNotificationService to set the userId. 

There's this other way that I thinks is more direct: 
Taking a look at the structure: 
```
 <Web3Provider network={process.env.REACT_APP_FLOW_ENV}>
          <NotificationModalProvider>
            <NotificationServiceProvider>
              <Router>
                <ErrorHandler>
                  <AppPages />
                </ErrorHandler>
              </Router>
            </NotificationServiceProvider>
          </NotificationModalProvider>
        </Web3Provider>
  ```
The Web3Provider is wrapping all components, so we can do inside the `NotificationServiceProvider`: 
```
import { useWebContext } from 'contexts/Web3';
.
.
.
const [notificationSettings, setNotificationSettings] = useState({
    walletId: '',
    email: '',
    communitySubscription: [],
    isUnsubscribedFromCommunityUpdates: false,
  });

 const { user: { addr }  } = useWebContext();  

useEffect(()=> {
  if(addr) {
   setNotificationSettings(prev => ({
      ...prev,
      walletId: addr,
    }))
  }
  if(!addr && notificationSettings.walletId !== ''){
    // cleaning state bc user got disconnected
      setNotificationSettings({
          walletId: '',
          email: '',
          communitySubscription: [],
         isUnsubscribedFromCommunityUpdates: false,  
        } <== all of this could be a static object since it's used in the initialization  too 
     )
   }
}, [addr])

```
Then we don't need to make SignInOutButton component aware of the existence of this logic of the `NotificationServiceProvider`. We are coupling directly NotificationServiceProvider with the Web3Context provider directly and not through the SignInOutButton. 

When a user connects/disconnects it causes the re-render of the components that are consuming through the context so this will make the useEffect to run and set/un-set the state in the `NotificationServiceProvider`
Let me know if you have any questions :)

Reply:
Yes this approach is way better! Updated!